### DISCUSSION TITLE
How to measure a cell width and height in a table or a grid ?

### DISCUSSION THREAD

**Discussion by debdagybra** [typst/typst#3943] [Q&A] [answered]
How can I get `width` and `height` of the cells of a `table` or a `grid` in context.

I tried the code below (in typst.app with version 0.11.0) but 
- for the whole table, the result is weird.
-  when I look at the children, the `columns` parameter of the `table` isn't used anymore and the result isn't what I want.

```{=typst}
#set page(
    width: 210mm, height: 297mm,
    margin: (left: 2cm, right: 2cm, top: 2cm, bottom: 2cm),
  )

#context [

    #let table_1 = table(
        columns: (1fr, 1fr),
        rect()[#lorem(3)], 
        rect()[#lorem(6)]
      )
      
    #table_1
    table size:         #measure(table_1) \
    left cell size:     #measure(table_1.children.at(0)) \
    right cell size:    #measure(table_1.children.at(1)) \

    #let table_2 = table(
        columns: (2fr, 1fr), // I changed first column to 2fr
        rect()[#lorem(3)], 
        rect()[#lorem(6)]
      )
      
    #table_2
    table size:        #measure(table_2) \
    left cell size:    #measure(table_2.children.at(0)) \
    #text(fill: red, "Not what I expected") 
    right cell size:   #measure(table_2.children.at(1)) \
]
```

![image](https://github.com/typst/typst/assets/10846318/7f1640f6-1de7-4c18-9b22-178705b6b69e)

I expected something like right cell size:  (width: 140pt, height: 34pt)

Do you have any ideas how I can do it ?

**Comment by HugoForrat** [comment, marked as answer]
This is an old post but I stumbled across it while having a similar issue, so maybe this can help someone.

The behaviour is explained by this line in [`measure`'s documentation](https://typst.app/docs/reference/layout/measure/)

> By default an infinite space is assumed, so the measured dimensions may not necessarily match the final dimensions of the content.

This isn't the case when you use fractional lengths for your table. The width
of the table will be constrained by the width of the page, and the widths of
the cell will be half the width of the page each for the first table, or two
third and one third of the width of the page for the second table.

If in your code, I replace the measurements by 

```typst
#layout(((width, height: _)) => [
	table size:         #measure(table_1, width: width) \
	left cell size:     #measure(table_1.children.at(0), width: width/2) \
	right cell size:    #measure(table_1.children.at(1), width: width/2) \
])
```

and

```typst
#layout(((width, height: _)) => [
	table size:         #measure(table_2, width: width) \
	left cell size:     #measure(table_2.children.at(0), width: (2/3) * width) \
	right cell size:    #measure(table_2.children.at(1), width: (1/3) * width) \
])
```
I get:
<img width="967" height="368" alt="out2" src="https://github.com/user-attachments/assets/53132c72-c971-4036-9177-b2e73cbf1449" />

**Comment by debdagybra** [comment]
thank you !

**Comment by tshort** [comment]
@HugoForrat's answer shows the width and height of cell content. Is there a way to get the positions of the cell corners? 

I'd like to draw on a table using the positions of the cell corners.
